### PR TITLE
[TW-1514] Flows typescript support

### PR DESCRIPTION
**Pull request by petewitcher-postman**
Created a hidden page for TypeScript support in Flows. Link to the page in beta (VPN required):

https://tw-1514-flows-typescript-support.learning.postman-beta.com/docs/postman-flows/concepts/flows-typescript-support/

This page is only accessible with the URL. It won't appear in searches.

### CODE REVIEW

**Review by MatthewPinsker-postman — APPROVED** [review]
LGTM!

**Review by jkonrath-postman — APPROVED** [review]

File: src/pages/docs/postman-flows/concepts/flows-typescript-support.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: "Typescript support in Postman Flows"
+updated: 2023-11-22
+early_access: true
+plan: beta
+---
+
+> Postman Flows TypeScript support is in beta.
+
+The **Evaluate** block in Postman Flows now supports TypeScript in addition to [Flows Query Langauge (FQL)](/docs/postman-flows/flows-query-language/introduction-to-fql/).
```
[the remark this answers is not in the document]
Comment (petewitcher-postman) [reply src/pages/docs/postman-flows/concepts/flows-typescript-support.md:10]: Thanks Daniel, included the If block here.

File: src/pages/docs/postman-flows/concepts/flows-typescript-support.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+---
+title: "Typescript support in Postman Flows"
+updated: 2023-11-22
+early_access: true
+plan: beta
+---
+
+> Postman Flows TypeScript support is in beta.
```
[the remark this answers is not in the document]
Comment (petewitcher-postman) [reply src/pages/docs/postman-flows/concepts/flows-typescript-support.md:8]: Done

File: src/pages/docs/postman-flows/concepts/flows-typescript-support.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Contents
+
+* [TypeScript compared to FQL](#typescript-compared-to-fql)
+* [Restrictions and notes](#restrictions-and-notes)
+* [How to use TypeScript in Postman Flows](#how-to-use-typescript-in-postman-flows)
+
+## TypeScript compared to FQL
+
+While FQL is tailored for querying and transforming JSON data within Postman Flows, TypeScript offers the robustness and flexibility of a full programming language. This makes TypeScript more suitable for complex logic, large-scale projects, and developers familiar with both JavaScript and TypeScript ecosystems. TypeScript's static typing also provides better predictability and error checking than FQL.
```
Comment (rabbah) [inline src/pages/docs/postman-flows/concepts/flows-typescript-support.md:20]: i suggest removing `TypeScript's static typing also provides better predictability and error checking than FQL.`
Rather, we can say that TypeScript may be more desirable for developers familiar with it and prefer to use it over FQL.

File: src/pages/docs/postman-flows/concepts/flows-typescript-support.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+## Restrictions and notes
+
+Scripts written with TypeScript in Postman Flows have the following restrictions:
+
+* Scripts can't import or require libraries.
+* Scripts don't have file system access.
+* The top level code must be an expression.
+
+TypeScript in Postman Flows uses a forked version of the CryptoJS library. This forked version uses an internal API for random number generation.
```
Comment (rabbah) [inline src/pages/docs/postman-flows/concepts/flows-typescript-support.md:30]: I would take this out "uses a forked version of the CryptoJS library". This is an implementation detail for us. 
Instead we should say that we support the following cryptographic functions which are useful for generating random numbers, encoding or decoding data, or verifying signed data.

@saswatds can we include the crypto library API that is available for us or link to the web standard (I forgot if you ended up with a hybrid approach or the web crypto standard).

File: src/pages/docs/postman-flows/concepts/flows-typescript-support.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Typescript support in Postman Flows"
+updated: 2023-12-15
+early_access: true
+plan: beta
+---
+
+Postman Flows now supports [TypeScript](https://www.typescriptlang.org/) in addition to [Flows Query Langauge (FQL)](/docs/postman-flows/flows-query-language/introduction-to-fql/) in the **Evaluate** and **If** blocks.
+
+<img src="https://assets.postman.com/postman-docs/v10/flows-typescript-v10-20.jpg" alt="TypeScript in an Evaluate block"/>
```
Comment (rabbah) [inline src/pages/docs/postman-flows/concepts/flows-typescript-support.md:10]: Can we include a screenshot or a simpler example - for example filtering an array or mapping values in an array? I think this example is too complex as the first intro example... even just using a simple expression and returning `hello typescript` might be better

We should use this example later down when we talk about the crypto library

Comment (petewitcher-postman) [reply src/pages/docs/postman-flows/concepts/flows-typescript-support.md:10]: Changed screenshot to filtering an array.

File: src/pages/docs/postman-flows/concepts/flows-typescript-support.md (hunk 3 above)
Comment (petewitcher-postman) [reply src/pages/docs/postman-flows/concepts/flows-typescript-support.md:20]: Updated paragraph to:

While FQL is tailored for querying and transforming JSON data within Postman Flows, TypeScript offers the robustness and flexibility of a full programming language. This makes TypeScript more suitable for complex logic and large-scale projects. Also, developers who are familiar with both JavaScript and TypeScript ecosystems may prefer using TypeScript over FQL.

File: src/pages/docs/postman-flows/concepts/flows-typescript-support.md (hunk 4 above)
Comment (petewitcher-postman) [reply src/pages/docs/postman-flows/concepts/flows-typescript-support.md:31]: @rabbah @saswatds We can add this change after we publish the Dec 15 release. Resolving this conversation for this PR.

**Pull request merged**